applications/rag/example_notebooks/rag-ray-ingest-with-kubernetes-docs.ipynb

```diff
@@ -0,0 +1,289 @@
+    "BATCH_SIZE = 100\n",
+    "CHUNK_SIZE = 1000  # text chunk sizes which will be converted to vector embeddings\n",
+    "CHUNK_OVERLAP = 10\n",
+    "VECTOR_DIMENSION = 384  # Embeddings size\n",
+    "\n",
+    "splitter = RecursiveCharacterTextSplitter(chunk_size=CHUNK_SIZE, chunk_overlap=CHUNK_OVERLAP, length_function=len)\n",
+    "embeddings_service = HuggingFaceEmbeddings(model_name=SENTENCE_TRANSFORMER_MODEL)\n",
+    "\n",
+    "def process_pdf(file_path):\n",
+    "    \"\"\"Loads, splits and embed a single PDF file.\"\"\"\n",
+    "    loader = PyPDFLoader(file_path)\n",
+    "    print(f\"Loading {file_path}\")\n",
+    "    pages = loader.load_and_split()\n",
+    "    \n",
+    "    splits = splitter.split_documents(pages)\n",
+    "\n",
+    "    chunks = []\n",
+    "    for split in splits:\n",
+    "        id = uuid.uuid4()\n",
+    "        page_content = split.page_content\n",
+    "        file_metadata = split.metadata\n",
+    "        embedded_document = embeddings_service.embed_query(page_content)\n",
+    "        split_data = {\n",
+    "            \"langchain_id\" : id,\n",
+    "            \"content\" : page_content,\n",
+    "            \"embedding\" : embedded_document,\n",
+    "            \"langchain_metadata\" : file_metadata\n",
+    "        }\n",
+    "        chunks.append(split_data)\n",
+    "    return chunks\n",
+    "\n",
+    "documents_file_paths = glob.glob(f\"{SHARED_DATASET_BASE_PATH}/PDFs/*.pdf\")\n",
+    "for file_path in documents_file_paths:\n",
+    "    processed_result = process_pdf(file_path)\n",
+    "    DBSession.bulk_insert_mappings(TextEmbedding, processed_result)\n",
+    "        \n",
+    "DBSession.commit()\n",
+    "print (\"end job\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6b9bc582-50cd-4d7c-b5c4-549626fd2349",
+   "metadata": {},
+   "source": [
+    "## Summiting the job into Ray Cluster:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d5b6acbe-5a14-4bc8-a4ca-58a6b3dd5391",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ray, time\n",
+    "from ray.job_submission import JobSubmissionClient\n",
+    "client = JobSubmissionClient(\"ray://ray-cluster-kuberay-head-svc:10001\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4eb8eae9-2a20-4c02-ac79-196942ae2783",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Port forward to the Ray dashboard and go to `localhost:8265` in a browser to see job status: kubectl port-forward -n <namespace> service/ray-cluster-kuberay-head-svc 8265:8265\n",
+    "import time\n",
+    "\n",
+    "start_time = time.time()\n",
+    "job_id = client.submit_job(\n",
+    "    entrypoint=\"python job.py\",\n",
+    "    # Path to the local directory that contains the entrypoint file.\n",
+    "    runtime_env={\n",
+    "        \"working_dir\": \"/home/jovyan/rag-app\", # upload the local working directory to ray workers\n",
```

Comment:
I wasn't sure so I took the same approach from https://github.com/GoogleCloudPlatform/ai-on-gke/blob/87d11d5499aae2f4a76950ebd38d396e702bbc77/applications/rag/example_notebooks/rag-kaggle-ray-sql-latest.ipynb#L292, there the working dir is setup as `/home/jovyan/rag-app\` is updated so just can be `/home/rag-app\`

Reply:
Regarding to this, I think we prefer the interactive pattern which `rag-kaggle-ray-sql-interactive.ipynb` followes. It would be great if we can follow the pattern in this notebook.